### ISSUE TITLE
Consider npm over yarn

### ISSUE THREAD

**Issue by jdvivar** [ing-bank/lion#242]
Hi

When Yarn was released in 2016 it offered 2 great features over npm: a **lock file** to make sure everyone (even build servers) were using exactly the same packages, and **workspaces**.

Since npm has now a mature **lock file** and since you're using Lerna for package management, what are the reasons to still use Yarn?

This is a genuine question, I'm sure there might be good reasons there that I'm missing :)

I was a hardcore Yarn user back in 2017, all my team went back to npm in 2018, but this project is from 2019.

Articles from devs experiences:
- https://iamturns.com/yarn-vs-npm-2018/
- https://medium.com/@vincentnewkirk/npm-vs-yarn-2019-e88757b17038

**Comment by jorenbroekema** [comment]
Hi

Apart from those two benefits:

- From what I can remember, when we switched to monorepo with lerna, we had some difficulties using NPM, whereas with yarn it worked instantly.
- It is also much faster even still (https://github.com/pnpm/benchmarks-of-javascript-package-managers), in most scenarios. Not twice as fast maybe as it used to be, but still substantially better.
- NPM's lock file is still reported to be buggy sometimes and it gets corrupted often, something I have faced a lot of times. And even if you commit the package-lock.json, sometimes it still unexpectedly changes when you do npm install, when you don't expect it. I wouldn't say npm has a mature lock file from personal experiences. For yarn at least it does what a lockfile is (imo) supposed to do.
- You have a point about lerna, I guess you would expect us to use yarn workspaces instead of lerna, if we committed to yarn. Then again, it works for us even with yarn, and I've heard good things about pnpm workspaces too, and currently a big downside for us with lerna is that we have a looooot of packages that depend on each other, which explains why we have so many releases. So I think experimenting with other tools here (like yarn / pnpm workspaces) might be cool. I think @daKmoR  probably has an opinion on this ? :)
- I know some people dislike the npm team and how they handle their community and collaboration, but I am not sure if that's a deal breaker, or even a thing, for people in our team.
  
All in all, I don't think we're very strict or opinionated about our preference of yarn currently, so personally I'm happy to discuss and look at alternatives or going back to npm if it turns out to be better now.
